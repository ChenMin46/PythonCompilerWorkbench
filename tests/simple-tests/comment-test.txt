# comment on top

x = 5 # this is a comment
  # an another comment
      # indented even more
y = 10 # yet another one
# oh yeahhhh

# one more
