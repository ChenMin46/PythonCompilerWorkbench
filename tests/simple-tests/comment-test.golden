# comment on top

x = 5 # this is a comment
  # an another comment
      # indented even more # with another comment inside
y = 10 # yet another one with triple hash ###
# oh yeahhhh

z = "a hash # within a string shouldn't be a comment"

# one more
===
Module
  body: [
    Assign >> 'x = 5'
      targets: [
        Name >> 'x'
          id: 'x'
          ctx: Store
      ]
      value: Num >> '5'
        n: 5
    Assign >> 'y = 10'
      targets: [
        Name >> 'y'
          id: 'y'
          ctx: Store
      ]
      value: Num >> '10'
        n: 10
    Assign >> 'z = "a hash # within a string shouldn\'t be a comment"'
      targets: [
        Name >> 'z'
          id: 'z'
          ctx: Store
      ]
      value: Str >> '"a hash # within a string shouldn\'t be a comment"'
        s: "a hash # within a string shouldn't be a comment"
  ]
--- JSON ---
{
  "contents": [
    {
      "id": "tid_1", 
      "name": "comment", 
      "value": "# comment on top"
    }, 
    "\n\n", 
    {
      "contents": [
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_2", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "5"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        " ", 
        {
          "id": "tid_3", 
          "name": "comment", 
          "value": "# this is a comment"
        }, 
        "\n  ", 
        {
          "id": "tid_4", 
          "name": "comment", 
          "value": "# an another comment"
        }, 
        "\n      ", 
        {
          "id": "tid_5", 
          "name": "comment", 
          "value": "# indented even more # with another comment inside"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "y"
            }, 
            " ", 
            {
              "id": "tid_6", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "10"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        " ", 
        {
          "id": "tid_7", 
          "name": "comment", 
          "value": "# yet another one with triple hash ###"
        }, 
        "\n", 
        {
          "id": "tid_8", 
          "name": "comment", 
          "value": "# oh yeahhhh"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "z"
            }, 
            " ", 
            {
              "id": "tid_9", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Str", 
              "value": "\"a hash # within a string shouldn't be a comment\""
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n\n", 
    {
      "id": "tid_10", 
      "name": "comment", 
      "value": "# one more"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
